README.md

```diff
@@ -67,33 +67,39 @@ cmake -G "Unix Makefiles" [OPTIONS] .. && make
 Replace `[OPTIONS]` with desired cmake options for your build.
 Ginkgo adds the following additional switches to control what is being built:
 
-*   `-DDEVEL_TOOLS={ON, OFF}` sets up the build system for development
+*   `-DGINKGO_DEVEL_TOOLS={ON, OFF}` sets up the build system for development
     (requires clang-format, will also download git-cmake-format),
     default is `ON`
-*   `-DBUILD_TESTS={ON, OFF}` builds Ginkgo's tests
+*   `-DGINKGO_BUILD_TESTS={ON, OFF}` builds Ginkgo's tests
     (will download googletest), default is `ON`
-*   `-DBUILD_BENCHMARKS={ON, OFF}` builds Ginkgo's benchmarks
+*   `-DGINKGO_BUILD_BENCHMARKS={ON, OFF}` builds Ginkgo's benchmarks
     (will download gflags and rapidjson), default is `ON`
-*   `-DBUILD_EXAMPLES={ON, OFF}` builds Ginkgo's examples, default is `ON`
-*   `-DBUILD_REFERENCE={ON, OFF}` build reference implementations of the
-    kernels, usefull for testing, default os `OFF`
-*   `-DBUILD_OMP={ON, OFF}` builds optimized OpenMP versions of the kernels,
+*   `-DGINKGO_BUILD_EXAMPLES={ON, OFF}` builds Ginkgo's examples, default is `ON`
+*   `-DGINKGO_BUILD_REFERENCE={ON, OFF}` build reference implementations of the
+    kernels, useful for testing, default is `OFF`
+*   `-DGINKGO_BUILD_OMP={ON, OFF}` builds optimized OpenMP versions of the kernels,
     default is `OFF`
-*   `-DBUILD_CUDA={ON, OFF}` builds optimized cuda versions of the kernels
+*   `-DGINKGO_BUILD_CUDA={ON, OFF}` builds optimized cuda versions of the kernels
     (requires CUDA), default is `OFF`
-*   `-DBUILD_DOC={ON, OFF}` creates an HTML version of Ginkgo's documentation
-    from inline comments in the code
-*   `-DSET_CUDA_HOST_COMPILER={ON, OFF}` instructs the build system to
-    explicitly set CUDA's host compiler to match the commpiler used to build the
+*   `-DGINKGO_BUILD_DOC={ON, OFF}` creates an HTML version of Ginkgo's documentation
+    from inline comments in the code. The default is `OFF`.
+*   `-DGINKGO_DOC_GENERATE_PDF={ON, OFF}` generates a PDF version of Ginkgo's
+    documentation from inline comments in the code. The default is `OFF`.
+*   `-DGINKGO_DOC_GENERATE_DEV={ON, OFF}` generates the developer version of
+    Ginkgo's documentation. The default is `OFF`.
+*   `-DGINKGO_SET_CUDA_HOST_COMPILER={ON, OFF}` instructs the build system to
+    explicitly set CUDA's host compiler to match the compiler used to build the
     the rest of the library (otherwise the nvcc toolchain uses its default host
     compiler). Setting this option may help if you're experiencing linking
     errors due to ABI incompatibilities. The default is `OFF`.
+*   `-DGINKGO_EXPORT_BUILD_DIR={ON, OFF}` adds the Ginkgo build directory to the
```

Comment:
Any reason this is  `OFF` by default ?

Reply:
This setting to me is mostly useful if Ginkgo is used as a subdirectory by another package. It could pollute a lot the package registry of everyone if it was `ON` by default and you would maybe need sometimes to clean `~/.cmake/packages/Ginkgo/` depending on what you want. If the user needs to explicitly activate it they should know what they are doing and what to expect out of it.

Reply:
Just to mention, setting this option to `ON` gives the ability to any library which uses Ginkgo as a subdirectory/subproject to find the build directory (after it was built) simply using `find_package(Ginkgo)`. This is why it is very useful when Ginkgo is used as a subdirectory.

Reply:
Final mention, this is actually set in #216 . I added to the README all options which weren't mentioned.